src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs

```diff
@@ -399,10 +438,25 @@ private IReadOnlyList<ObjectPropertyInitializer> GetPropertyInitializers(IReadOn
             return null;
         }
 
+        private HashSet<string> GetParentPropertyNames()
+        {
+            return EnumerateHierarchy()
+                .Skip(1)
+                .SelectMany(type => type.Properties)
+                .Select(p => p.Declaration.Name)
+                .ToHashSet();
+        }
+
         protected override IEnumerable<ObjectTypeProperty> BuildProperties()
         {
+            var existingPropertyNames = GetParentPropertyNames();
             foreach (var field in Fields)
-                yield return new ObjectTypeProperty(field, Fields.GetInputByField(field));
+            {
+                var property = new ObjectTypeProperty(field, Fields.GetInputByField(field));
+                if (existingPropertyNames.Contains(property.Declaration.Name))
```

Comment:
To confirm, if one property whose name also exists in both ParentPropertyNames but with different type, which one will you use?

Reply:
we use the inherited one.
Overriding a property and changing its type (type narrow down) is not supported.
C# does not support type narrow down, therefore to support it, we need to design what does it look like first.
Issue tracking that: https://github.com/Azure/autorest.csharp/issues/4096
Now we just ignore all attempts to override the property in base.

For example:
```
model Base {
	a: string | int;
}

model Derived extends Base {
	a: string;
}
```
We will ignore the `a` defined in `Derived` (this piece of code did this).

When we need to support [type narrow down](https://github.com/Azure/autorest.csharp/issues/4096), we will go over the design first, and then change this part to implement it.